### PR TITLE
Refactoring adding services

### PR DESCRIPTION
**Pull request by Suracki**
- updated SafetyAlertsModel to be static and handle load/save automatically; controllers now request confirmation from the model that it has been loaded before running commands to allow any errors in the data load to be reported to the user
- Added services for all controllers & moved dependencies out of controllers
- *Controller and *Service now extend BaseController or BaseService classes containing shared dependencies
- all tests updated to work with new structure

### CODE REVIEW

File: src/main/java/com/safetynet/alerts/logic/updaters/ResultModel.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    public ResultModel(SafetyAlertsModel model, Boolean bool){
+        this.model = model;
+        this.bool = bool;
+    }
+
+    public SafetyAlertsModel getModel() {
+        return model;
+    }
+
+    public Boolean getBool() {
```
Comment (fpatelm) [inline src/main/java/com/safetynet/alerts/logic/updaters/ResultModel.java]: what this bool represents on the result model? I would rename it to something more meaningfull

File: src/main/java/com/safetynet/alerts/presentation/controller/BaseController.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+import com.safetynet.alerts.logging.LogHandlerTiny;
+import com.safetynet.alerts.presentation.model.SafetyAlertsModel;
+import org.springframework.beans.factory.annotation.Autowired;
+
+public class BaseController {
+
+    @Autowired
+    LogHandlerTiny logHandler;
+
+    @Autowired
```
Comment (fpatelm) [inline src/main/java/com/safetynet/alerts/presentation/controller/BaseController.java]: I would remove this (`@Autowired SafetyAlertsModel safetyAlertsModel;`) from BaseController and keep it in BaseService. And on the BaseService add a method eg. `bool isDataLoaded()` that can be used by any service that is used in the controller

File: src/main/java/com/safetynet/alerts/presentation/controller/FirestationController.java [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+     * @return Json string & HttpStatus.CREATED if successful
+     */
+    @PostMapping("/firestation")
+    public ResponseEntity<String> addEntity(@RequestParam("address") String address, @RequestParam("station") int station){
+        //Log reqquest
+        logHandler.setLogger("FirestationController");
+        logHandler.logRequest("POST","/firestation", new String[] {address, String.valueOf(station)});
+
+        //confirm data is loaded
+        if (!safetyAlertsModel.isDataLoaded()){
```
Comment (fpatelm) [inline src/main/java/com/safetynet/alerts/presentation/controller/FirestationController.java]: I would change it to `!firestationService.isDataLoaded()`

Apply this same approach for all controllers that are calling the `!safetyAlertsModel.isDataLoaded()`

File: src/main/java/com/safetynet/alerts/presentation/controller/MedicalRecordController.java [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+ *
+ * Includes POST/PUT/DELETE
+ */
+@RestController
+public class MedicalRecordController extends BaseController {
+
+    @Autowired
+    private MedicalRecordService medicalRecordService;
+
+    private String stringArrayToString(String[] stringArray) {
```
Comment (fpatelm) [inline src/main/java/com/safetynet/alerts/presentation/controller/MedicalRecordController.java]: this method seems used by several controllers, if it is the same implementation, move it to BaseController

File: src/main/java/com/safetynet/alerts/logic/service/BaseService.java [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+@Service
+public class BaseService {
+
+    @Autowired
+    LogHandlerTiny logHandler;
+    @Autowired
+    SafetyAlertsModel model;
+    @Autowired
+    ModelObjectFinder finder;
+
```
Comment (fpatelm) [inline src/main/java/com/safetynet/alerts/logic/service/BaseService.java:18]: add
`bool isDataLoaded(){
return this.model.isDataLoaded();
}`

**Review by fpatelm — CHANGES_REQUESTED** [review]
changes in comments

File: src/main/java/com/safetynet/alerts/logic/updaters/ResultModel.java (hunk 1 above)
Comment (Suracki) [reply src/main/java/com/safetynet/alerts/logic/updaters/ResultModel.java]: boolean represents if the operation was successful - updated now

File: src/main/java/com/safetynet/alerts/presentation/controller/BaseController.java (hunk 2 above)
Comment (Suracki) [reply src/main/java/com/safetynet/alerts/presentation/controller/BaseController.java]: makes sense, done for all controllers

File: src/main/java/com/safetynet/alerts/presentation/controller/FirestationController.java (hunk 3 above)
Comment (Suracki) [reply src/main/java/com/safetynet/alerts/presentation/controller/FirestationController.java]: does make more sense this way with the model removed from the controllers and only in the services; updated

File: src/main/java/com/safetynet/alerts/presentation/controller/MedicalRecordController.java (hunk 4 above)
Comment (Suracki) [reply src/main/java/com/safetynet/alerts/presentation/controller/MedicalRecordController.java]: done, and also done with intArrayToStringArray as it is used for a similar purpose

File: src/main/java/com/safetynet/alerts/logic/service/BaseService.java (hunk 5 above)
Comment (Suracki) [reply src/main/java/com/safetynet/alerts/logic/service/BaseService.java:18]: done

**Pull request merged**